go-controller/pkg/ovn/kubevirt.go

```diff
@@ -46,18 +53,261 @@ func (oc *DefaultNetworkController) deleteDHCPOptions(pod *kapi.Pod) error {
+			policyVMLabel, ok := policy.ExternalIDs[kubevirtv1.VirtualMachineNameLabel]
+			return ok && policyVMLabel == pod.Labels[kubevirtv1.VirtualMachineNameLabel]
+		}); err != nil {
+			return fmt.Errorf("failed configuring pod routing when deleting stale policies for pod %s/%s: %v", pod.Namespace, pod.Name, err)
+		}
+		if err := libovsdbops.DeleteLogicalRouterStaticRoutesWithPredicate(oc.nbClient, types.OVNClusterRouter, func(route *nbdb.LogicalRouterStaticRoute) bool {
+			routeVMLabel, ok := route.ExternalIDs[kubevirtv1.VirtualMachineNameLabel]
+			return ok && routeVMLabel == pod.Labels[kubevirtv1.VirtualMachineNameLabel]
+		}); err != nil {
+			return fmt.Errorf("failed configuring pod routing when deleting stale static routes for pod %s/%s: %v", pod.Namespace, pod.Name, err)
+		}
+		return nil
+	}
+
+	lrpAddresses, err := util.GetLRPAddrs(oc.nbClient, types.GWRouterToJoinSwitchPrefix+types.GWRouterPrefix+pod.Spec.NodeName)
+	if err != nil {
+		return fmt.Errorf("failed configuring pod routing when reading LRP addresses: %v", err)
+	}
+
+	// For IC this route will match the cluster CIDR and route traffic to
+	// the gw, the VM's live migrated IP will match there when they
+	// are migrated at a node not owning the VM subnet
+	if config.OVNKubernetesFeature.EnableInterconnect {
+		for _, clusterSubnet := range config.Default.ClusterSubnets {
+			nodeGwAddress, err := util.MatchFirstIPNetFamily(utilnet.IPFamilyOfCIDR(clusterSubnet.CIDR) == utilnet.IPv6, lrpAddresses)
+			if err != nil {
+				return err
+			}
+			clusterSubnetRoute := nbdb.LogicalRouterStaticRoute{
+				IPPrefix: clusterSubnet.CIDR.String(),
+				Nexthop:  nodeGwAddress.IP.String(),
+				Policy:   &nbdb.LogicalRouterStaticRoutePolicySrcIP,
+				ExternalIDs: map[string]string{
+					kubevirt.OvnZoneExternalIDKey:      kubevirt.OvnLocalZone,
+					kubevirtv1.VirtualMachineNameLabel: kubevirt.AllVMs,
+				},
+			}
+			if err := libovsdbops.CreateOrReplaceLogicalRouterStaticRouteWithPredicate(oc.nbClient, types.OVNClusterRouter, &clusterSubnetRoute, func(item *nbdb.LogicalRouterStaticRoute) bool {
+				matches := item.IPPrefix == clusterSubnetRoute.IPPrefix && item.Nexthop == clusterSubnetRoute.Nexthop && item.Policy != nil && *item.Policy == *clusterSubnetRoute.Policy
+				return matches
+			}); err != nil {
+				return fmt.Errorf("failed adding static route: %v", err)
+			}
+		}
+	}
+
+	for _, podIP := range podAnnotation.IPs {
+		podAddress := podIP.IP.String()
+		// In case of one global zone a policy is needed to route traffic
+		// to the gateway
+		if !config.OVNKubernetesFeature.EnableInterconnect {
+			ipFamily := utilnet.IPFamilyOfCIDR(podIP)
+			nodeGwAddress, err := util.MatchFirstIPNetFamily(ipFamily == utilnet.IPv6, lrpAddresses)
+			if err != nil {
+				return err
+			}
+			match := fmt.Sprintf("ip%s.src == %s", ipFamily, podAddress)
+			egressPolicy := nbdb.LogicalRouterPolicy{
```

Comment:
so this is to route a migrated pod's egress traffic back to the GR on the node where the pod lives, correct?

Reply:
yep

Reply:
Can we add a comment here like: // adds a policy so that a migrated pods egress traffic will be routed to the local GR where it now resides